### DISCUSSION TITLE
Allow specific tests to not run in parallel with each other but to run in parallel with other tests

### DISCUSSION THREAD

**Discussion by benmenachem** [nunit/nunit#4657] [Q&A] [answered]
Hello, 
I'm using SpecFlow & Nunit as testing framework and I have a few feature files that run in parallel 
currently all tests run in parallel besides tests that that are marked as [NonParallelizable]
I was looking for a way to limit certain features to run in parallel with each other but to be able to run in parallel with other features. 
For example assume I have 4 feature files: Feature1, Feature2, Feature3 and Feature4 and assume Feature1 can't run in parallel with Feature2
So using [NonParallelizable] would allow Feature1 and Feature2 to not run in parallel with any feature file. 
I'd like to know if there's an option to limit Feature1 from running in parallel with Feature2 but for each one of them to be able to run in parallel with Feature3 and Feature4 

Thanks in advance for any assistance!

**Comment by OsirisTerje** [comment, marked as answer]
I am sorry, but there is no direct support for this in NUnit.  

You may achieve this though by placing them in different assemblies but having Feature 1 and Feature 2 in the same assembly, marking them as not `NonParallelizable`, and then having 3 and 4 in another one, marking them either way.  Using `dotnet test` and its way to run assemblies in parallel, it would run the the assembly with F.1 and F.2 in parallel with the one containing F.3 and F.4.  You can then set the `Parallelizable` attribute for that second assembly, thus making F.3 and F.4 tests to run in parallel with each other if you like.

The setting for running in parallel in dotnet test is the MaxCpuCount property. The default is 1, which means no parallel, but set it to any other number and it will do parallel. Setting it to 0 will do as many as you have cores. See details here:  
https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#runconfiguration-element

You include a runsettings file like:   `dotnet test -s .runsettings`, assuming you have named it as `.runsettings`.

**Comment by benmenachem** [comment]
Thank you for the reply!
